core/domain/takeout_service_test.py

```diff
@@ -876,6 +929,180 @@ def test_export_data_for_full_user_trivial_is_correct(self):
+            clazz
+            for clazz in self._get_model_classes()
+            if not clazz.__name__ in self.BASE_CLASSES
+        ]
+
+        # Iterate over models and test export policies.
+        for model in all_models:
+            export_method = model.get_model_association_to_user()
+            export_policy = model.get_export_policy()
+            renamed_export_keys = model.get_field_names_for_takeout()
+            exported_property_names = []
+            property_used_as_key_for_takeout_dict = None
+            for property_name in model._properties: # pylint: disable=protected-access
+                if (export_policy[property_name] ==
+                        base_models.EXPORT_POLICY.EXPORTED):
+                    if property_name in renamed_export_keys:
+                        exported_property_names.append(
+                            renamed_export_keys[property_name]
+                        )
+                    else:
+                        exported_property_names.append(property_name)
+                elif (export_policy[property_name] ==
+                      base_models
+                      .EXPORT_POLICY.EXPORTED_AS_KEY_FOR_TAKEOUT_DICT):
+                    property_used_as_key_for_takeout_dict = property_name
+
+            if (export_method ==
+                    base_models
+                    .MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER):
+                self.assertEqual(len(exported_property_names), 0)
+            elif (export_method ==
+                  base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER):
+                exported_data = model.export_data(self.USER_ID_1)
+                self.assertEqual(
+                    sorted([
+                        python_utils.UNICODE(key)
+                        for key in exported_data.keys()]),
+                    sorted(exported_property_names)
+                )
+            elif (export_method ==
+                  base_models
+                  .MODEL_ASSOCIATION_TO_USER
+                  .ONE_INSTANCE_SHARED_ACROSS_USERS):
+                self.assertIsNotNone(
+                    model.get_field_name_mapping_to_takeout_keys)
+                exported_data = model.export_data(self.USER_ID_1)
+                field_mapping = model.get_field_name_mapping_to_takeout_keys()
+                self.assertEqual(
+                    sorted(exported_property_names),
+                    sorted(field_mapping.keys())
+                )
+                self.assertEqual(
+                    sorted(exported_data.keys()),
+                    sorted(field_mapping.values())
+                )
+            elif (export_method ==
+                  base_models
+                  .MODEL_ASSOCIATION_TO_USER.MULTIPLE_INSTANCES_PER_USER):
+                exported_data = model.export_data(self.USER_ID_1)
+                for model_id in exported_data.keys():
+                    # If we are using a property as a Takeout key.
```

Comment:
For this export type, won't it always be the case that we are?

Reply:
No. See above https://github.com/oppia/oppia/pull/10944#discussion_r519018010